### ISSUE TITLE
[8088] Philips P3105/NMS9100 cannot select processor frequency above 4.77Mhz

### ISSUE THREAD

**Issue by maximus105** [86Box/86Box#3830]
### What happened?

On this machine, when installing an 8088 or NEC V20 processor, the processor frequency always remains 4.77Mhz, even if you select 7.16Mhz, 8Mhz, 10Mhz, 12Mhz, 16Mhz.

### Configuration file

```ini
[Machine]
machine = p3105
cpu_use_dynarec = 0
time_sync = local
cpu_enable_fpu = 1
cpu_family = 8088
cpu_speed = 7159092
cpu_multi = 1
fpu_type = 8087
fpu_softfloat = 0
mem_size = 512

[CGA]
display_type = 0
composite_type = 0
rgb_type = 0
snow_enabled = 0

[General]
vid_renderer = qt_opengl
window_remember = 1
vid_resize = 1
force_43 = 1
vid_cga_contrast = 1
confirm_exit = 0
video_fullscreen_first = 0
confirm_reset = 0
confirm_save = 0

[Video]
gfxcard = superega

[Input devices]
mouse_type = msserial
joystick_type = 2axis_2button
joystick_0_nr = 0
joystick_1_nr = 0

[Sound]
fm_driver = nuked

[Hard disks]
hdd_01_parameters = 17, 4, 306, 0, ide
hdd_01_fn = images/DOS.img
hdd_01_speed = ramdisk
hdd_01_ide_channel = 0:0

[Storage controllers]
hdc = xtide_acculogic
cartridge_01_fn = G:/Temp/86Box/
cartridge_02_fn = G:/Temp/86Box/

[Other peripherals]
postcard_enabled = 1

[Monitor #1]
window_coordinates = 412, 31, 1736, 1292

[Monitor #2]
window_coordinates = 412, 31, 1736, 1292

[Floppy and CD-ROM drives]
fdd_02_type = none

[Microsoft Serial Mouse]
port = 0
buttons = 4
```

### Operating system

Windows 10 Pro x64 v22H2

### CPU

Core i9-10900K

### 86Box version

v4.1 Build 5207

### Build architecture

Windows - x86 (32-bit)

### Build type

- [ ] New recompiler
- [ ] Debug build

### Download source

Official website (Jenkins, GitHub)

**Comment by guestuser0001** [comment]
Can you send the hard-disk drive image file?

**Comment by maximus105** [comment]
[DOS.zip](https://github.com/86Box/86Box/files/13364317/DOS.zip)

**Comment by guestuser0001** [comment]
When doing a Landmark Speed test, the CPU frequency shows (*4.948MHz*), and the CPU speed performs so slow, 2.09MHz...
While the FPU speed was a quite slower, 4.44MHz...
![image](https://github.com/86Box/86Box/assets/147709293/6ab79b1e-d2c2-4eb0-a438-f0a3e77ee134)
It just said (This computer performs like a 2MHz AT (must be *XT*), with a 4MHz 80287 (must be *8087*))...

**Comment by OBattler** [comment]
How are quirks of the Landmark Speed test 86Box bugs?

**Comment by OBattler** [comment]
Closing - the setting is saved correctly so it's being used. I'm not sure where you're seeing the wrong speed because I can assure you, the correct speed is in fact being used.

**Comment by maximus105** [comment]
You misunderstood the bug description. The bug is not that the frequency is incorrect, but that in 86Box for this machine for the 8088 or NEC V20 processor, the frequencies available for selection are 7.16MHz, 8MHz, 10MHz, 12MHz, 16MHz. But the processor frequency in 86Box will always be 4.77MHz, when choosing any of these frequencies. If this machine only supports 4.77 MHz, then you need to remove all other frequencies from the 86Box selection list. And if this machine supports all these frequencies, then the bug needs to be fixed. So that, for example, when choosing a frequency of 16 MHz, the machine boots at this frequency. It is better to measure the processor frequency using the System Information program from the Norton Utilities v6.0 package for MS-DOS.
